website/content/docs/concepts/integrated-storage/index.mdx

```diff
@@ -168,15 +178,92 @@ node2    node2.vault.local:8201    follower    true
 node3    node3.vault.local:8201    leader      true
 ```
 
-## Server-to-Server Communication
+## Integrated Storage and TLS
 
-Once nodes are joined to one another they begin to communicate using mTLS over
-Vault's cluster port. The cluster port defaults to `8201`. The TLS information
-is exchanged at join time and is rotated on a cadence.
+We've glossed over some details in the above sections on bootstrapping clusters.
+The instructions are sufficient for most cases, but some users have run into
+problems when using auto-join and TLS in conjunction with things like auto-scaling.
+The issue is that go-discover on most platforms returns IPs (not hostnames), and
+because the IPs aren't knowable in advance, the TLS certificates used to secure
+the Vault API port don't contain these IPs in their IP SANs.
 
-A requirement for integrated storage is that the
-[cluster_addr](/docs/concepts/ha#per-node-cluster-address) configuration option
-is set. This allows Vault to assign an address to the node ID at join time.
+### Vault networking recap
+
+Before we explore solutions to this problem, let's recapitulate how Vault nodes
+speak to one another.
+
+Vault exposes two TCP ports: the API port and the cluster port.
+
+The API port is where clients send their Vault HTTP requests.
+
+For a single-node Vault cluster you don't worry about a cluster port as it won't be used.
+
+When you have multiple nodes you also need a cluster port.  This is used by Vault
+nodes to issue RPCs to one another, e.g. to forward requests from a standby node
+to the active node, or when Raft is in use, to handle leader election and
+replication of stored data.
+
+The cluster port is secured using a TLS certificate that the Vault active node
+generates internally.  It's clear how this can work when not using integrated
+storage: every node has at least read access to storage, so once the active
+node has persisted the certificate, the standby nodes can fetch it, and all
+agree on how cluster traffic should be encrypted.
+
+It's less clear how this works with integrated storage, as there is a chicken
+and egg problem.  Nodes don't have a shared view of storage until the raft
+cluster has been formed, but we're trying to form the raft cluster!  To solve
+this problem, a Vault node must speak to another Vault node using the API port
+instead of the cluster port.  This is currently the only situation in which
+OSS Vault does this (Vault Enterprise also does something similar when setting
+up replication.)
+
+* node2 wants to join the cluster, so issues challenge API request to existing member node1
+* node1 replies to challenge request with (1) an encrypted random UUID and (2) seal config
+* node2 must decrypt UUID using seal; if using auto-unseal can do it directly, if using shamir must wait for user to provide enough unseal keys to perform decryption
```

Comment:
```suggestion
* node2 must decrypt UUID using seal. If using auto-unseal it can do so directly, and if using shamir it must wait for the user to provide enough unseal keys to perform decryption
```